Mathlib/FieldTheory/LinearDisjoint.lean

```diff
@@ -365,6 +365,139 @@ theorem of_finrank_sup [FiniteDimensional F A] [FiniteDimensional F B]
     (H : finrank F ↥(A ⊔ B) = finrank F A * finrank F B) : A.LinearDisjoint B :=
   linearDisjoint_iff'.2 <| .of_finrank_sup_of_free (by rwa [← sup_toSubalgebra_of_left])
 
+/-- If `A` and `L` are linearly disjoint, one of them is algebraic, then `[L(A) : L] = [A : F]`. -/
```

Comment:
I think this is currently unused; would you prove `[E : L] = [E : L[A]] * [A : F]` too?

Reply:
The `adjoin_rank_eq_rank_left_of_isAlgebraic` is used in the golfed proof of `sepDegree_eq_of_isPurelyInseparable_of_isSeparable`. As the `[E : L] = [E : L(A)] * [A : F]` it's just `rank_mul_rank` followed by `rw` (in this case `E / L(A) / L` is indeed a field extension tower in Lean, unlike the `E / L(A) / A`).